app/web/features/auth/useAuthStore.test.ts

```diff
@@ -36,6 +39,35 @@ describe("usePersistedState hook", () => {
   });
 });
 
+describe("useSessionStorage hook", () => {
+  it("uses a default value", () => {
+    const defaultValue = "Test string";
+    const { result } = renderHook(() => useSessionStorage("key", defaultValue));
+    expect(result.current[0]).toBe(defaultValue);
+  });
+
+  it("saves then loads a value", () => {
+    const value = { test: "Test string" };
+    const { result } = renderHook(() => useSessionStorage("key", { test: "" }));
+    expect(result.current[0]).toStrictEqual({ test: "" });
+    act(() => result.current[1](value));
+    expect(result.current[0]).toStrictEqual(value);
+    expect(sessionStorage.getItem("key")).toBe(JSON.stringify(value));
+    const { result: result2 } = renderHook(() =>
+      useSessionStorage("key", { test: "" })
+    );
+    expect(result2.current[0]).toStrictEqual(value);
+  });
+
+  it("clears a value", () => {
+    const { result } = renderHook(() => useSessionStorage("key", { test: "" }));
+    expect(result.current[0]).toStrictEqual({ test: "" });
+    act(() => result.current[2]());
+    expect(result.current[0]).toStrictEqual(undefined);
+    expect(sessionStorage.getItem("key")).toBe(null);
+  });
+});
```

Comment:
I know you've wrote these based on the existing tests... but I think it'd be clearer for readability if we destructure `result.current` further and call the named destructured function - i.e.

```typescript
// instead of this (what the heck is `result.current[1]`? bit hard to tell without having to
// jump back into the implementation...
act(() => result.current[1](value));

// do...
const [state, setState, clearState] = result.current;
act(() => setState(value))
```
etc

(The classic of reading back old code when we were "more naive" a year ago and be like "why did we allow this crap" 🙈 )

Reply:
I think it actually makes some sense to do it this way since destructuring `current` would make the `state` value stale 🤔

Reply:
Very good point, and having to destructure each time might be too verbose. Hmm yeah maybe let's leave it like this then 👍